### ISSUE TITLE
Feature request: \cctab_gsave:N, \g_tmpa_cctab, \g_tmpb_cctab

### ISSUE THREAD

**Issue by jlaurens** [latex3/latex3#1089]
That would be another name for LuaTeX's `\savecatcodetable` but I guess something very different for pdfTeX.

**Comment by josephwright** [comment]
Could you give an example of a use case? We've not really used catcode tables much yet.

**Comment by jlaurens** [comment]
My own use case:
```latex
\begin{verbatim}
Line 1
Line 2 |\textbf{This is bold text}|
\end{verbatim}
``` 
The part between the `|` is typeset with the real catcode regime, including 3rd party's changes.

**Comment by jlaurens** [comment]
The workflow:

1. save the current catcode regime in one of the scratch variables
2. make any catcode modification, either local or global
3. typeset material either in saved or modified regime
4. restore the catcode regime

This is interesting when catcode modifications are known at run time only, or when they are allowed to 3rd party.
This may also save one group level.

As `\savecatcodetable` acts globally in LuaTeX, I added a `g` qualifier.

An alternate solution is to save and restore the catcode of only a subset of characters. But that would require a different data model and more work.

**Comment by josephwright** [comment]
@blefloch suggests `\cctab_gsave_current:N` - agreed to implement.

**Comment by josephwright** [comment]
Addressed by ba396bed98d8fb4c9700cd404c00b15f98152af7
